### ISSUE TITLE
Support DRM-encrypted VODs

### ISSUE THREAD

**Issue by birme** [Eyevinn/channel-engine#224]

**Comment by birme** [comment]
Support verified with HLS + Widevine (CMAF). Fairplay yet to be verified

**Comment by birme** [comment]
Picked up by the daily-backlog-pr implementation phase but could not be implemented as-is — returning to Ready pending specification.

The issue body is empty (no description, acceptance criteria, or proposed API), and the headline feature is already substantially shipped: `examples/drm.ts` is a working HLS + Widevine (CMAF) VOD reference, the README lists "Support for DRM encrypted HLS VODs (in beta)", and the 2023 maintainer note records "Support verified with HLS + Widevine (CMAF). Fairplay yet to be verified."

To make this actionable a maintainer needs to specify:
1. What scope of "DRM" is still outstanding vs. already shipped (e.g. is this now purely FairPlay/CBCS verification?).
2. Concrete acceptance criteria — which DRM systems, HLS vs DASH/CMAF, CENC vs CBCS, muxed vs demuxed.
3. Whether code changes are needed at all, or just verification/docs and lifting the "beta" caveat.
4. Test assets and a target license-server setup for the outstanding case.

Suggest re-scoping to the specific remaining gap (likely FairPlay) or routing to `architect` for breakdown.

🤖 Automated via Channel Engine Dev daily-backlog-pr skill

**Comment by birme** [comment]
Picked up by the daily-backlog-pr automation run (2026-09-04) but not implemented: this issue has an empty body and no acceptance criteria, and "Support DRM-encrypted VODs" is too broad to translate into a single automated PR safely (DRM support spans key delivery, manifest signaling, and packaging decisions that need a human-authored spec first).

Leaving it in **Ready** and unassigned. Recommend a maintainer break it down into concrete, testable sub-issues (e.g. supported DRM systems/key-delivery, manifest signaling, config surface) before the automation can act on it.

🤖 Automated via Channel Engine Dev daily-backlog-pr skill

**Comment by birme** [comment]
Picked up by today's daily-backlog-pr run but **not implementable as a single scoped PR**: the issue body is empty and "Support DRM-encrypted VODs" is a broad tracking umbrella rather than one concrete deliverable (which is why prior automation runs also could not action it).

The concrete, specced slice of this work is now tracked in **#401** ("reconcile DRM support table with the verified examples/drm.ts path and add regression coverage"), which is Ready on the board this run. Per the architect's triage, #224 can be closed as superseded once #401 lands. Leaving #224 in Ready and unassigned for a maintainer to close/keep as the umbrella.

**Comment by birme** [comment]
Picked up by the daily-backlog-pr run but not actionable as a single implementable PR: this issue has an empty body and no acceptance criteria — it reads as a 2023 tracking epic rather than a scoped unit of work. The concrete, verifiable DRM follow-up is already scoped in #401 (reconcile the DRM support table against the `examples/drm.ts` path and add regression coverage).

Unassigning and moving back to Ready. A human maintainer should either close this epic in favour of #401 (and any remaining scoped sub-issues) or add explicit acceptance criteria so an engineer can implement it unambiguously.

**Comment by birme** [comment]
Picked up by the Channel Engine Dev daily-backlog-pr run (Phase 2) but **not actionable as-is** — returning to Ready for maintainer attention.

The issue body is empty (opened 2023) and DRM is no longer greenfield: the repo already ships a beta DRM feature — `examples/drm.ts` (Widevine HLS+CMAF via ezdrm), `spec/engine/drm_spec.ts`, and the README support table marking **HLS+CMAF = "Yes (Widevine)"**, **HLS+TS = "No"**. channel-engine delegates DRM to `@eyevinn/hls-vodtolive` (EXT-X-KEY/EXT-X-MAP passthrough); it does no key delivery or re-encryption itself. So this ticket is a broad umbrella overlapping already-shipped work with no bounded increment.

To become Ready, a maintainer needs to specify:
1. Which DRM systems beyond the existing Widevine beta (FairPlay? PlayReady? multi-DRM/CENC?).
2. Which source formats — specifically HLS+TS DRM (currently "No"), or promoting the CMAF+Widevine beta to GA?
3. Config surface & responsibility — should the engine own key-delivery/session-key config, or keep delegating passthrough to `hls-vodtolive`?
4. Acceptance criteria & testability, given CI has no licence server or DRM-capable player.

Recommend routing to `architect` to break this into concrete, format-scoped sub-issues, or closing it as superseded by the shipped beta. No code was written.

**Comment by birme** [comment]
**daily-backlog-pr (Phase 2): can't implement — no specification.**

This item is in *Ready* but its description is empty — "Support DRM-encrypted VODs" with no body. DRM/VOD-encryption support (key delivery, DRM system(s), manifest/segment signaling, and the integration seam across `channel-engine` and `docker-fast`) can't be scoped or implemented without concrete requirements and acceptance criteria.

Routing back for **architect breakdown** into scoped, implementable sub-issues before an engineer picks it up. Left in *Ready*, unassigned.

**Comment by birme** [comment]
Not implemented this run — under-specified for direct implementation. This is a broad "Support DRM-encrypted VODs" epic with no acceptance criteria or scoped deliverable in the body, so routing it straight to an engineer would be guesswork. Recommend re-triage/breakdown by the architect into concrete, independently-implementable sub-issues (e.g. per DRM system / manifest signaling / key delivery) before it re-enters implementation. Left in Ready.

**Comment by birme** [comment]
Picked up by the daily-backlog-pr automation. This issue has an **empty body** (just the 2023 title) and, on investigation, DRM-encrypted VOD support already exists on `master`:

- `examples/drm.ts` — working reference wiring the engine to Widevine-encrypted HLS+CMAF VODs.
- `README.md` support table — HLS+CMAF → DRM: Yes (Widevine); "Widevine DRM encrypted HLS+CMAF VODs (in beta)".
- `spec/engine/drm_spec.ts` — regression coverage added via #401 / PR #406 (EXT-X-KEY + CMAF EXT-X-MAP pass-through into video and demuxed-audio renditions).

There is no spec in #224 to distinguish it from what already ships, so the automation is **not** implementing anything here. Recommend a human **close this as already satisfied**. If a stakeholder specifically wants FairPlay/PlayReady, HLS+TS DRM, or DRM+subtitles, please open a **new, separately-specced** issue rather than re-scoping this empty one.

Leaving in Ready for a maintainer decision; not auto-closing since the automation isn't scoped to close a feature request on its own judgement.

**Comment by birme** [comment]
daily-backlog-pr (2026-09-15): not actionable this run. This issue has an **empty body** — no scope, acceptance criteria, or reproduction — so an engineer cannot implement it without guessing. Additionally, DRM support already exists in the codebase (`examples/drm.ts`, and the DRM support table was reconciled/regression-covered under #401). A human maintainer should either specify the concrete remaining DRM gap (with acceptance criteria) or close this as already-addressed. Leaving in Ready, unassigned.

**Comment by birme** [comment]
NEEDS HUMAN INPUT: (daily-backlog-pr 2026-09-16) not implementable by automation — needs a maintainer decision, not another engineer pass.

This is an empty-body 2023 umbrella epic, and the headline capability already ships on `master`: `examples/drm.ts` (Widevine HLS+CMAF VOD reference), the README support table (HLS+CMAF → "Yes (Widevine)", in beta), and regression coverage in `spec/engine/drm_spec.ts` landed via #401/PR #406. Every automated run from 2026-09-03 through 2026-09-15 reached the same conclusion and returned it to Ready.

There is no bounded, testable increment left in *this* ticket. A maintainer should either:
- **close it as already satisfied**, or
- open a **new, separately-specced** issue for any concrete remaining gap (FairPlay/PlayReady, HLS+TS DRM, DRM+subtitles) with acceptance criteria and test assets.

Marking this with an explicit NEEDS HUMAN INPUT marker so future automated runs skip re-attempting it until a human weighs in. Left in Ready, unassigned.
